plasmapy/atomic/particle_class.py

```diff
@@ -0,0 +1,701 @@
+            f"this attribute is not available.")
+
+        self._isotope_errmsg = (
+            f"The particle '{self.particle}' does not have an "
+            f"isotope specified, so this attribute is not available.")
+
+        self._ion_errmsg = (
+            f"The particle '{self.particle}' is not an ion, so this"
+            f"attribute is not available.")
+
+    def __repr__(self) -> str:
+        r"""Returns a string of the call that would recreate this object."""
+        return f'Particle("{self.particle}")'
+
+    def __str__(self) -> str:
+        r"""Returns a string of the particle symbol."""
+        return f"{self.particle}"
+
+    def __eq__(self, other) -> bool:
+        r"""Returns True when comparing two Particle objects that correspond
+        to the same particle, and False when the two objects differ."""
+
+        if not isinstance(other, Particle):
+            return False
+
+        try:
+            if self.particle == other.particle:
+                return True
+            else:
+                return False
+        except Exception:
+            return False
+
+    def __ne__(self, other) -> bool:
+        r"""Returns True when the two objects differ, and False when
+        comparing two Particle objects that correspond to the same particle."""
+        return not self.__eq__(other)
+
+    @property
+    def particle(self) -> str:
+        r"""Returns the particle symbol."""
+        return self._attributes['particle']
+
+    @property
+    def element(self) -> Optional[str]:
+        r"""Returns the atomic symbol if the particle corresponds to an
+        element, and None otherwise."""
+        return self._attributes['element']
+
+    @property
+    def isotope(self) -> Optional[str]:
+        r"""Returns the isotope symbol if the particle corresponds to an
+        isotope, and None otherwise."""
+        return self._attributes['isotope']
+
+    @property
+    def ion(self) -> Optional[str]:
+        r"""Returns the ion symbol if the particle corresponds to an ion,
+        and None otherwise."""
+        return self._attributes['ion']
+
+    @property
+    def element_name(self) -> str:
+        r"""Returns the name of the element corresponding to this particle,
+        or raises an InvalidElementError if the particle does not correspond
+        to an element."""
+        if not self.element:
+            raise InvalidElementError(self._element_errmsg)
+        return self._attributes['element name']
+
+    @property
+    def integer_charge(self) -> int:
+        r"""Returns the integer charge of the particle, or raises a ChargeError
+        if the charge has not been specified."""
+        if self._attributes['integer charge'] is None:
+            raise ChargeError(
+                f"The charge of particle {self.particle} has not been "
+                f"specified.")
+        return self._attributes['integer charge']
+
+    @property
+    def charge(self) -> u.C:
```

Comment:
You are mixing units and types as return annotations here, I suggest you decorate the function with `u.quantity_input`  when you annotate with units rather than types.

Reply:
I had been wondering about the best way to annotate methods that return `Quantity` objects while still keeping track of units, so it's great to know that `u.quantity_input` can handle this.  Thanks!